File: Java/_387_First_Unique_Character_in_a_String/Solution.java
class Solution {
    public int firstUniqChar(String s) {
        if (s == null || s.equals("")) return -1;
		char[] arr = s.toCharArray();
		int[] count = new int[256];				
		for (int i = 0; i < arr.length; i++) {
			count[arr[i]]++;
		}
		for (int i = 0; i < arr.length; i++) {
			if (count[arr[i]] == 1) return i; 
		}
		return -1;
    }
}

File: Python/_387_First_Unique_Character_in_a_String/solution.py
class Solution:
    def firstUniqChar(self, s):
        """
        :type s: str
        :rtype: int
        """
        count = [0] * 128
        for i in range(0, len(s)):
            count[ord(s[i]) - ord('a')] += 1
        for i in range(0, len(s)):
            if count[ord(s[i]) - ord('a')] == 1:
                return i
        return -1
